### PR TITLE
Added support for relations

### PR DESCRIPTION
**Pull request by AviadPepperi**
Refactored code.
Renamed MyService -> MonitorSettingsService.
Created Utils module and Relations service.

### CODE REVIEW

File: server-side/installation.ts [hunk 1]
```diff
@@ -25,10 +26,9 @@ exports.install = async (client: Client, request: Request) => {
         let successUsageMonitor = true;
 
         client.AddonUUID = "7e15d4cc-55a7-4128-a9fe-0e877ba90069";
```
Comment (AviadPepperi) [inline server-side/installation.ts:28]: Can be deleted?

File: server-side/installation.ts [hunk 2]
```diff
@@ -99,100 +99,104 @@ exports.install = async (client: Client, request: Request) => {
...
             success: false,
-            errorMessage: ('message' in err) ? err.message : 'Cannot install HealthMonitorAddon. Unknown Error Occured',
+            errorMessage: errorMessage,
         };
     }
 };
 
 exports.uninstall = async (client: Client, request: Request) => {
     try {
         client.AddonUUID = "7e15d4cc-55a7-4128-a9fe-0e877ba90069";
```
Comment (AviadPepperi) [inline server-side/installation.ts:147]: Can be deleted?

File: server-side/installation.ts [hunk 3]
```diff
@@ -233,22 +239,21 @@ exports.uninstall = async (client: Client, request: Request) => {
 exports.upgrade = async (client: Client, request: Request) => {
 
     let success = true;
-    let errorMessage = '';
     let resultObject = {};
 
     client.AddonUUID = "7e15d4cc-55a7-4128-a9fe-0e877ba90069";
```
Comment (AviadPepperi) [inline server-side/installation.ts:244]: Can be deleted?

File: server-side/api.ts [hunk 4]
```diff
@@ -8,53 +10,53 @@ const sleep = (milliseconds) => {
...
+    "PASSED-JOB-LIMIT": { "Message": 'Distributor passed the job limit', "Color": "FF0000" },
+    "TIMEOUT-GET-UDT": { "Message": 'Get udt call timeout', "Color": "FF0000" },
+    "TIMEOUT-SYNC": { "Message": 'Sync call timeout', "Color": "FF0000" },
+    "TIMEOUT-SYNC-FAILED-TEST": { "Message": 'sync_failed test got timeout', "Color": "FF0000" }
 };
 
 //#region health monitor api
 export async function sync_failed(client: Client, request: Request) {
     console.log('HealthMonitorAddon start SyncFailed test');
     client.AddonUUID = "7e15d4cc-55a7-4128-a9fe-0e877ba90069";
```
Comment (AviadPepperi) [inline server-side/api.ts:33]: Can be deleted?

File: server-side/api.ts [hunk 5]
```diff
@@ -90,14 +92,16 @@ export async function job_limit_reached(client: Client, request: Request) {
     console.log('HealthMonitorAddon start JobLimitReached test');
     try {
         client.AddonUUID = "7e15d4cc-55a7-4128-a9fe-0e877ba90069";
```
Comment (AviadPepperi) [inline server-side/api.ts:94]: Can be deleted?

File: server-side/api.ts [hunk 6]
```diff
@@ -107,14 +111,16 @@ export async function addon_limit_reached(client: Client, request: Request) {
     console.log('HealthMonitorAddon start AddonLimitReached test');
     try {
         client.AddonUUID = "7e15d4cc-55a7-4128-a9fe-0e877ba90069";
```
Comment (AviadPepperi) [inline server-side/api.ts:113]: Can be deleted?

File: server-side/api.ts [hunk 7]
```diff
@@ -123,31 +129,33 @@ export async function job_execution_failed(client: Client, request: Request) {
...
-            errorMessage:errorMessage
+            success: false,
+            errorMessage: errorMessage
         };
     }
 };
 
-export async function health_monitor_settings(client: Client, request: Request){
+export async function health_monitor_settings(client: Client, request: Request) {
     client.AddonUUID = "7e15d4cc-55a7-4128-a9fe-0e877ba90069";
```
Comment (AviadPepperi) [inline server-side/api.ts:149]: Can be deleted?

File: server-side/installation.ts (hunk 2 above)
Comment (nofarpepperi) [reply server-side/installation.ts:147]: yes

File: server-side/installation.ts (hunk 3 above)
Comment (nofarpepperi) [reply server-side/installation.ts:244]: yes

File: server-side/utils.service.ts [hunk 8]
```diff
@@ -0,0 +1,16 @@
+export module Utils {
+
+    export function GetErrorDetailsSafe(error: unknown, field = "message", defaultValue = 'Unknown Error Occured'): string {
+
+        try {
+            if (error instanceof Error) {
+                if (field in error) {
+                    return error.message;
```
Comment (nofarpepperi) [inline server-side/utils.service.ts:8]: use error[field] instead of error.message

File: server-side/api.ts [hunk 9]
```diff
@@ -618,128 +631,136 @@ export async function JobLimitReachedTest(service) {
...
-            if (report.length==0){
-                const reportMessage= "No new errors were found since " + intervalUTCDate + ".";
-                ReportErrorCloudWatch(await GetDistributorCache(service, monitorSettings), "JOB-EXECUTION-REPORT", "JOB-EXECUTION-FAILED", innerMessage);
-                console.log("HealthMonitorAddon, JobExecutionFailedTest finish");
-                return {
-                    success:true, 
-                    resultObject:reportMessage
-                };
-            }
+        // Relations entry point
```
Comment (nofarpepperi) [inline server-side/api.ts:735]: add to comment there here you called to addons that have relation to health monitor

File: server-side/relations.service.ts [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+
+export interface HosteesData {
+    Code: string,
+    Type: string,
+    GeneralErrorMessage: string,
+    InternalErrors?: HosteesInternalErrors[],
+    [key: string]: any
+}
+
+function instanceOfHosteesData(object: any): object is HosteesData {
```
Comment (nofarpepperi) [inline server-side/relations.service.ts:21]: change function name to IsInstanceOfHosteesData

File: server-side/utils.service.ts (hunk 8 above)
Comment (AviadPepperi) [reply server-side/utils.service.ts:8]: #Facepalm

Comment (AviadPepperi) [reply server-side/utils.service.ts:8]: changed to use [field]

File: server-side/installation.ts (hunk 2 above)
Comment (AviadPepperi) [reply server-side/installation.ts]: Deleted

File: server-side/installation.ts (hunk 1 above)
Comment (AviadPepperi) [reply server-side/installation.ts]: Deleted

File: server-side/api.ts (hunk 4 above)
Comment (AviadPepperi) [reply server-side/api.ts]: Deleted

File: server-side/api.ts (hunk 5 above)
Comment (AviadPepperi) [reply server-side/api.ts]: Deleted

File: server-side/api.ts (hunk 7 above)
Comment (AviadPepperi) [reply server-side/api.ts]: Deleted

File: server-side/api-success-monitor.ts [hunk 11]
```diff
@@ -1,11 +1,12 @@
-import MyService from './my.service'
+import MonitorSettingsService from './monitor-settings.service'
```
[the remark this answers is not in the document]
Comment (nofarpepperi) [reply server-side/api-success-monitor.ts]: yes, please remove the function and the comments that called it:
in upgrade function all the comments of: install UsageMonitor if not installed yet from version 2.0,
in install function all the comments of: install PepperiUsageMonitor code job

**Review by nofarpepperi — APPROVED** [review]

**Pull request merged**